dbms/src/Interpreters/Join.cpp

```diff
@@ -2301,4 +2421,18 @@ BlockInputStreamPtr Join::createStreamWithNonJoinedRows(const Block & left_sampl
     return std::make_shared<NonJoinedBlockInputStream>(*this, left_sample_block, index, step, max_block_size);
 }
 
+void ProbeProcessInfo::resetBlock(Block && block_)
+{
+    block = std::move(block_);
+    start_row = 0;
+    end_row = 0;
+    // If the probe block size is greater than max_block_size, we will set max_block_size to the probe block size to avoid some unnecessary split.
+    max_block_size = std::max(max_block_size, block.rows());
```

Comment:
set `all_rows_joined_finish` to false here?

Reply:
done.